src/Tests/HelixTasks/SDKCustomCreateXUnitWorkItems.cs

```diff
@@ -0,0 +1,152 @@
+        /// collates logged errors in order to determine the success of HelixWorkItem creation per
+        /// provided xUnit project data.
+        /// </summary>
+        /// <returns>A boolean value indicating the success of HelixWorkItem creation per provided xUnit project data.</returns>
+        public override bool Execute()
+        {
+            ExecuteAsync().GetAwaiter().GetResult();
+            return !Log.HasLoggedErrors;
+        }
+
+        /// <summary>
+        /// The asynchronous execution method for this MSBuild task which verifies the integrity of required properties
+        /// and validates their formatting, specifically determining whether the provided xUnit project data have a 
+        /// one-to-one mapping. It then creates this mapping before asynchronously preparing the HelixWorkItem TaskItem
+        /// objects via the PrepareWorkItem method.
+        /// </summary>
+        /// <returns></returns>
+        private async Task ExecuteAsync()
+        {
+            XUnitWorkItems = (await Task.WhenAll(XUnitProjects.Select(PrepareWorkItem)))
+                .SelectMany(i => i)
+                .Where(wi => wi != null)
+                .ToArray();
+            return;
+        }
+
+        /// <summary>
+        /// Prepares HelixWorkItem given xUnit project information.
+        /// </summary>
+        /// <param name="publishPath">The non-relative path to the publish directory.</param>
+        /// <returns>An ITaskItem instance representing the prepared HelixWorkItem.</returns>
+        private async Task<List<ITaskItem>> PrepareWorkItem(ITaskItem xunitProject)
+        {
+            // Forces this task to run asynchronously
+            await Task.Yield();
+
+            if (!xunitProject.GetRequiredMetadata(Log, "PublishDirectory", out string publishDirectory))
+            {
+                return null;
+            }
+            if (!xunitProject.GetRequiredMetadata(Log, "TargetPath", out string targetPath))
+            {
+                return null;
+            }
+            if (!xunitProject.GetRequiredMetadata(Log, "RuntimeTargetFramework", out string runtimeTargetFramework))
+            {
+                return null;
+            }
+
+            xunitProject.TryGetMetadata("Arguments", out string arguments);
+
+            string assemblyName = Path.GetFileName(targetPath);
+            if (!runtimeTargetFramework.Contains("core"))
+            {
+                throw new NotImplementedException("does not support non core runtime target");
+            }
+
+            string driver = $"{PathToDotnet} exec ";
+
+            // On mac due to https://github.com/dotnet/sdk/issues/3923, we run against workitem directory
+            // but on Windows, if we running against working item diretory, we would hit long path.
+            string testExecutionDirectory = IsPosixShell ? "-testExecutionDirectory $TestExecutionDirectory" : "-testExecutionDirectory %TestExecutionDirectory%";
```

Comment:
We should enable long paths on the Helix machines.  Hopefully that would mean we don't have long path issues.

Reply:
We had several meeting about. And we have this request since last year. So far no plan.

Reply:
When was this meeting? Long paths are required by practically every repository now including `dotnet/runtime`. At this point the expectation should be long paths are enabled everywhere, it shouldn't be the exception. Even DevDiv has added long path support to their build machines. Need to get Helix in line with this policy.

Reply:
We discussed this with @wli3 because their payloads hit this immediately. We made the decision to not enable long paths because we still support Windows 7 and Windows 8 where long paths aren't supported.

Reply:
Can't we enable long paths everywhere else though? Basically why can't the machine images, for everything that isn't Windows 7 or 8, have long paths enabled?